org.eclipse.jdt.ui/ui/org/eclipse/jdt/ui/text/java/correction/ASTRewriteCorrectionProposal.java

```diff
@@ -32,6 +32,18 @@
  * @since 3.8
  */
 public class ASTRewriteCorrectionProposal extends CUCorrectionProposal {
+	/**
+	 * Constructs an AST rewrite correction proposal.
+	 * @param delegate The delegate instance
+	 * @param image the image that is displayed for this proposal or <code>null</code> if no image
+	 *            is desired
+	 * @since 3.32
+	 */
+	public ASTRewriteCorrectionProposal(ASTRewriteCorrectionProposalCore delegate, Image image) {
```

Comment:
I get a warning in my Eclipse saying that ASTRewriteCorrectionProposalCore is non-API parameter type.  Is this something that needs tweaking in the MANIFEST.MF for the packages??

Reply:
Yeah, I'm not really sure how to handle this tbh.  

The Core versions are clearly not API and are internal. The API ones allow you to create one out of core. I don't particularly see this as a bug. It's just a constructor that most downstream consumers won't be able to access.

Reply:
There is a commented usage filter that can be added.  I guess that it is the best way to do it.